pkg/controller/bastion/actuator_reconcile.go

```diff
@@ -0,0 +1,319 @@
+	}
+
+	externalIP := &networkInterfaces[0].AccessConfigs[0].NatIP
+
+	if ingress := addressToIngress(&instance.Name, internalIP); ingress != nil {
+		endpoints.private = ingress
+	}
+
+	// GCP does not automatically assign a public dns name to the instance (in contrast to e.g. AWS).
+	// As we provide an externalIP to connect to the bastion, having a public dns name would just be an alternative way to connect to the bastion.
+	// Out of this reason, we spare the effort to create a PTR record (see https://cloud.google.com/compute/docs/instances/create-ptr-record#api) just for the sake of having it.
+	if ingress := addressToIngress(nil, externalIP); ingress != nil {
+		endpoints.public = ingress
+	}
+
+	return endpoints, nil
+}
+
+// IngressReady returns true if either an IP or a hostname or both are set.
+func IngressReady(ingress *corev1.LoadBalancerIngress) bool {
+	return ingress != nil && (ingress.Hostname != "" || ingress.IP != "")
+}
+
+// addressToIngress converts the IP address into a
+// corev1.LoadBalancerIngress resource. If both arguments are nil, then
+// nil is returned.
+func addressToIngress(dnsName *string, ipAddress *string) *corev1.LoadBalancerIngress {
+	var ingress *corev1.LoadBalancerIngress
+
+	if ipAddress != nil || dnsName != nil {
+		ingress = &corev1.LoadBalancerIngress{}
+		if dnsName != nil {
+			ingress.Hostname = *dnsName
+		}
+
+		if ipAddress != nil {
+			ingress.IP = *ipAddress
+		}
+	}
+
+	return ingress
+}
+
+func ensureDisk(ctx context.Context, gcpclient gcpclient.Interface, opt *Options) error {
+	disk, err := getDisk(ctx, gcpclient, opt)
+	if disk != nil || err != nil {
+		return err
+	}
+
+	logger.Info("create new bastion compute instance disk")
+	disk = diskDefine(opt.Zone, opt.DiskName)
+	_, err = gcpclient.Disks().Insert(opt.ProjectID, opt.Zone, disk).Context(ctx).Do()
+	if err != nil {
+		return fmt.Errorf("failed to create compute instance disk: %w", err)
+	}
+
+	disk, err = getDisk(ctx, gcpclient, opt)
+	if disk != nil || err != nil {
+		return err
+	}
+
+	return fmt.Errorf("failed to get (create) compute instance disk: %w", err)
+}
+
+func computeInstanceDefine(opt *Options, userData []byte) *compute.Instance {
+	return &compute.Instance{
+		Disks:              disksDefine(opt),
+		DeletionProtection: false,
+		Description:        "Bastion Instance",
+		Name:               opt.BastionInstanceName,
+		Zone:               opt.Zone,
+		MachineType:        machineTypeDefine(opt),
+		NetworkInterfaces:  networkInterfacesDefine(opt),
+		Tags:               &compute.Tags{Items: []string{opt.BastionInstanceName}},
+		Metadata:           &compute.Metadata{Items: metadataItemsDefine(userData)},
+	}
+}
+
+func metadataItemsDefine(userData []byte) []*compute.MetadataItems {
+	return []*compute.MetadataItems{
+		{
+			Key:   "startup-script",
+			Value: pointer.StringPtr(string(userData)),
+		},
+		{
+			Key:   "block-project-ssh-keys",
+			Value: pointer.StringPtr(string("TRUE")),
```

Comment:
redundant type conversion

```suggestion
			Value: pointer.StringPtr("TRUE"),
```

Reply:
fixed